jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/InputStreamContentSource.java

```diff
@@ -79,7 +81,7 @@ public Content.Chunk read()
         try
         {
             ByteBuffer buffer = streamBuffer.getByteBuffer();
-            int read = inputStream.read(buffer.array(), buffer.arrayOffset(), buffer.capacity());
+            int read = fillBufferFromInputStream(inputStream, buffer);
```

Comment:
In the old implementation everything was local and private, so we could assume the `ByteBuffer` was from heap, etc.
Now that we are calling an overridable method, we should avoid assumptions about how the `ByteBuffer` is.
Can we modify the signature of this method to take a `byte[]` instead, so there are no assumptions?

Reply:
That's a very good suggestion, let me do it.